src/cloudflare.rs

```diff
@@ -117,18 +118,34 @@ impl fmt::Display for ClError {
     }
 }
 
+fn apply_auth_to_request(
+    request_builder: reqwest::RequestBuilder,
+    email: String,
+    key: String,
+    token: String
+) -> reqwest::RequestBuilder {
```

Comment:
This can be written a bit better

```rust
fn apply_auth_to_request(
    request_builder: reqwest::RequestBuilder,
    email: String,
    key: String,
    token: String,
) -> reqwest::RequestBuilder {
    if !token.is_empty() {
        request_builder.bearer_auth(token)
    } else {
        request_builder
            .header("X-Auth-Email", email)
            .header("X-Auth-Key", key)
    }
}
```

Hmm wish I had a bit better design so that it wouldn't be necessary to pass in so many parameters (but you're good 👍 )

Reply:
Yeah my first attempt was passing in the config object (if that's the right terminology for Rust!) but the need to support the use of the CloudflareClient object too made me switch to multiple parameters. I'm not quite ready for writing a Rust function with a parameter that can be one of multiple types - no idea if the idea even makes any sense in the language so I figured this would be good enough for the time being :-)

Thanks for the Rust tips. I've made the change as you suggested.